src/screens/DiscoveryScreenV2/elements/DiscoveryTab.js

```diff
@@ -1,44 +1,62 @@
 import * as React from 'react';
 import {Dimensions, Keyboard, Pressable, ScrollView, StyleSheet, Text, View} from 'react-native';
 
+import {useRoute} from '@react-navigation/core';
 import {colors} from '../../../utils/colors';
 import {fonts, normalizeFontSize} from '../../../utils/fonts';
+import {setNavbarTitle} from '../../../context/actions/setMyProfileAction';
+import {Context} from '../../../context';
 
-const {width} = Dimensions.get('screen');
+const windowWidth = Dimensions.get('window').width;
+
+const DiscoveryTab = ({onChangeScreen, selectedScreen = 0, tabs}) => {
+  const [, dispatchNavbar] = React.useContext(Context).profile;
+  const route = useRoute();
 
-const DiscoveryTab = ({onChangeScreen, selectedScreen = 0}) => {
   const handleTabOnClicked = React.useCallback((index) => {
     Keyboard.dismiss();
     onChangeScreen(index);
   }, []);
 
-  const tabs = ['Users', 'Communities', 'Domains', 'News'];
+  const handleChangeTitle = (index = 0) => {
+    const title = ['Search Users', 'Your Communities', 'Your Domains'][index] || '';
+    setNavbarTitle(title, dispatchNavbar);
+  };
+
+  React.useEffect(() => {
+    if (route.name === 'Followings') handleChangeTitle();
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useEffect` hook is missing dependencies. It will only run once when the component mounts, which might not be the intended behavior. If you want it to run whenever `route.name` or `handleChangeTitle` changes, include them in the dependency array.
```diff
-   React.useEffect(() => {
-     if (route.name === 'Followings') handleChangeTitle();
-   }, []);

+   React.useEffect(() => {
+     if (route.name === 'Followings') handleChangeTitle();
+   }, [route.name, handleChangeTitle]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->